ui/raidboss/data/07-dt/trial/zelenia-ex.ts

```diff
@@ -1,14 +1,638 @@
+            break;
+          case bloomTileFlags.grey:
+            newState = 'grey';
+            break;
+        }
+        data.tileState[getTileNameFromLocation(matches.location)] = newState;
+      },
+    },
+    {
+      id: 'ZeleniaEx Phase Tracker',
+      type: 'StartsUsing',
+      netRegex: {
+        id: [
+          'A8AD',
+          'A8B5',
+          'A8CD',
+          'A8B9',
+          'AA14',
+          'AA15',
+          'A8C1',
+          'AA16',
+          'A8E8',
+          'AA17',
+          'AA18',
+        ],
+        capture: true,
+      },
+      suppressSeconds: 5,
+      run: (data, matches) => {
+        switch (matches.id) {
+          case 'A8AD': // Escelons' Fall (happens 3 times, only check first for phase1)
+            if (data.phase === 'phase1')
+              data.phase = 'escelon1';
+            break;
+          case 'A8B5': // Blessed Barricade
+            data.phase = 'adds';
+            break;
+          case 'A8CD': // Perfumed Quietus
+            data.phase = 'phase2';
+            break;
+          case 'A8B9': // Roseblood Bloom (happens twice, check for bloom 6)
+            data.phase = 'bloom1';
+            break;
+          case 'AA14': // Roseblood: 2nd Bloom
+            data.phase = 'bloom2';
+            break;
+          case 'AA15': // Roseblood: 3rd Bloom
+            data.phase = 'bloom3';
+            break;
+          case 'A8C1': // Explosion (happens 2 times, bloom 3 => escelon 2, bloom 6 ignore)
+            if (data.phase === 'bloom3')
+              data.phase = 'escelon2';
+            break;
+          case 'AA16': // Roseblood: 4th Bloom
+            data.phase = 'bloom4';
+            break;
+          case 'A8E8': // Alexandrian Banish III
+            data.phase = 'escelon3';
+            break;
+          case 'AA17': // Roseblood: 5th Bloom
+            data.phase = 'bloom5';
+            break;
+          case 'AA18': // Roseblood: 6th Bloom
+            data.phase = 'bloom6';
+            break;
+        }
+      },
+    },
+    {
+      id: 'ZeleniaEx Escelon Bait Collect',
+      type: 'GainsEffect',
+      // count: 2F6 = near, 2F7 = far
+      netRegex: { effectId: 'B9A', count: ['2F6', '2F7'] },
+      preRun: (data, matches) =>
+        data.escelonFallBaits.push(matches.count === '2F6' ? 'near' : 'far'),
+    },
+    {
+      id: 'ZeleniaEx Escelon Bait Cleanup',
+      type: 'GainsEffect',
+      // count: 2F6 = near, 2F7 = far
+      netRegex: { effectId: 'B9A', count: ['2F6', '2F7'], capture: false },
+      delaySeconds: 30,
+      suppressSeconds: 30,
+      run: (data) => data.escelonFallBaits = [],
+    },
+    {
+      id: 'ZeleniaEx Escelon Bait',
+      type: 'GainsEffect',
+      // count: 2F6 = near, 2F7 = far
+      netRegex: { effectId: 'B9A', count: ['2F6', '2F7'], capture: false },
+      durationSeconds: 19,
+      suppressSeconds: (data) => data.escelonFallBaits.length > 1 ? 20 : 0,
+      infoText: (data, _matches, output) => {
```

Comment:
Instead of doing a single infoText with a long duration, what do you think about instead doing:
 - an initial `infoText` to indicate whether the first bait is close or far (e.g. `(close baits first)`), and
 - depending on the pattern, firing a `alertText` to `Swap` after the 2nd hit (for ABBA) or after the 1st & 3rd hits (for AABB)?  
 
For strats that require the same role group to take the first bait whether in or out, this would give the necessary positioning info; and doing swap calls when they are contextually relevant might be more helpful.

Reply:
I'm trying to avoid super complicated logic here because this trigger is getting used for all 3 Escelon sets where timing can vary slightly, e.g. in Esc 2 due to your role being the one trapped in the donuts, or in Esc3 because you have to swap sides for the room cleave.

Reply:
The timing variations on Esc2 and Esc3 would, I think, only affect when the baited cleaves actually begin relative to the half-room cleaves or the donuts.  But in terms of the firing sequence, once it begins, I don't think there's any timing variation that would affect calls to swap?  But perhaps I'm misremembering.  Will try to jump in tomorrow and see.

(I'm also fine with writing something up as a follow-up PR if you'd rather land this as-is.)